blockchain/blockindex.go

```diff
@@ -228,6 +306,7 @@ type blockIndex struct {
 	// separate mutex.
 	db          database.DB
 	chainParams *chaincfg.Params
+	bestHeader  *blockNode
```

Comment:
Avoid adding bestHeader for now since this isn't needed for `ProcessBlockHeader()`. This and all associated code should be removed in this PR.